### DISCUSSION TITLE
How to use :class in x-spread?

### DISCUSSION THREAD

**Discussion by maxsite** [alpinejs/alpine#899] [1. Help]
There is such a code:

`<div x-data="{show: false}" @click="show = !show" :class="{'t-red': show}">toggle class click</div>`

I want to remake it on  x-spread

```
<div x-data="toggleInit('t-red')" x-spread="trigger">toggle class click</div>

<script>
function toggleInit(a) {
    return {
        show: false,
        
        trigger: {
            ['@click']() {
                this.show = !this.show;
            },
            
            [':class']() {
               console.log(this.show);
               console.log(a);
               
               return {a: this.show}; // not work
            },
        },
    }
}
</script>
```

What am I doing wrong?

**Comment by SimoTod** [comment]
Have you tried using ['x-bind:class']?

**Comment by KevinBatdorf** [comment]
Seems to work for me. What's version of Alpine are you using?

Can you make a demo?

**Comment by maxsite** [comment]
Alpine last version.

While I do this:

```
function toggleClick(a) {
    return {
        show: false,
        
        trigger: {
            ['@click']() {
                this.show = !this.show;
            },
            
            [':class']() {
                
                // work
                if (this.show) return a;
                return '';

                // return {a: this.show}; // not work
            },
        },
    }
}
```

**Comment by KevinBatdorf** [comment]
This works: https://codepen.io/KevinBatdorf/pen/RwGPwjd

I changed it to show red text instead when you click.

**Comment by maxsite** [comment]
The question is whether it is possible to use this:

`return {a: this.show};`

**Reply by KevinBatdorf** [reply]
The answer is yes, the class named "a" will be added depending on the state of the `show` property. 

I updated the above demo to demonstrate this exactly.


Maybe you're expecting `a` to not be a class name but a variable that contains a class name?

**Comment by maxsite** [comment]
I want to pass the class name as a function parameter.

**Reply by SimoTod** [reply]
Not using that syntax. In javascript, that one is the name of the key.

Something like this would work
```
[':class']() {
    const classes = {}
    classes[a] = this.show
    return classes;
},
```

Note, if i's a single value, this is probably easier to read

```
[':class']() {
    return this.show ? a : '';
},
```

**Reply by HugoDF** [reply]
This is newish syntax so watch out if you're supporting IE11, but this does what you want as well:

```
[':class']() {
    return {
        [a]: this.show
    }
},
```
See [computed property names](https://ui.dev/computed-property-names/)

**Comment by maxsite** [comment]
I'm understood, thank you. :)
